homeassistant/components/sonos/media_player.py

```diff
@@ -849,13 +874,13 @@ async def _async_handle_group_event(event):
         if event and not hasattr(event, "zone_player_uui_ds_in_group"):
             return
 
-        self.hass.add_job(_async_handle_group_event(event))
+        return _async_handle_group_event(event)
 
-    def update_content(self, event=None):
+    def async_update_content(self, event=None):
         """Update information about available content."""
         if event and "favorites_update_id" in event.variables:
-            self._set_favorites()
-            self.schedule_update_ha_state()
+            self.hass.async_add_job(self._set_favorites)
```

Comment:
We shouldn't use `hass.async_add_job` if we know what the target is.

Reply:
This is being called back from `pysonos` which can only take a callback so we can't `await` here.

I could decorate it with `@callback` though but its never called from Home Assistant

Reply:
Resolved in https://github.com/home-assistant/core/pull/48684 (but docker hub is broken at the moment so the CI is likely to fail)